### PR TITLE
Display correct ssl error message

### PR DESCRIPTION
**Pull request by mueller-ma**
The cause of the error a CertificateException and its cause is
CertPathValidatorException, so we need to call .getCause twice.

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java [hunk 1]
```diff
@@ -133,13 +133,13 @@ public void onFailure(Call call, int statusCode, Headers headers, byte[] respons
                 message = getString(R.string.error_unable_to_resolve_hostname);
             } else if (error instanceof SSLHandshakeException) {
                 // if ssl exception, check for some common problems
-                if (error.getCause() instanceof CertPathValidatorException) {
+                if (error.getCause().getCause() instanceof CertPathValidatorException) {
                     message = getString(R.string.error_certificate_not_trusted);
-                } else if (error.getCause() instanceof CertificateExpiredException) {
+                } else if (error.getCause().getCause() instanceof CertificateExpiredException) {
```
Comment (FlorianSW) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: When testing against https://expired.badssl.com the CertificateExpiredException is even one level under this:
e.getCause().getCause().getCause()

However, don't we've a better way of doing it? E.g. having a loop which calls getCause() until the returned object is not the same as the current one and then using this one for these checks, here?

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java]: The first implementation had .toString.equals(CertificateExpiredException). I'm not sure if thats better.

**Comment by mueller-ma** [comment]
Untrusted root and expired exceptions work now, however https://revoked.badssl.com/ gives a 404.

File: mobile/src/main/java/org/openhab/habdroid/util/Util.java [hunk 2]
```diff
@@ -146,4 +146,14 @@ public static void setActivityTheme(@NonNull final Activity activity, String the
         }
         activity.setTheme(themeRes);
     }
+
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/util/Util.java]: Why not pass cause as `Class<?> causeType` and compare `error.getCause()` to that? Saves the instance creations...

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/util/Util.java]: Yes, thats exactly what I'm currently trying. How can I call the method then?

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/util/Util.java]: Nevermind. got it

**Comment by FlorianSW** [comment]
Sorry for not repsonding anymore :( However, the change looks sane now! :)

**Pull request merged**